snuba/query/parser/exceptions.py

```diff
@@ -22,3 +22,7 @@ class CyclicAliasException(ValidationException):
 
 class AliasShadowingException(ValidationException):
     pass
+
+
+class FunctionValidationException(ValidationException):
+    pass
```

Comment:
I wouldn't want to pollute the exceptions, but... If we're going to have a lot of these would it be worth-while to attach `expr` to this (and possibly to an intermediate base `QueryValidationException`) and then implement `__str__` so we can count on `str(exception)` to send back consistently useful messages such as:

"Illegal function `function` at `function(blah)`"?  This would help us in the future if we moved toward structured logging and would move that exception string code you have in FunctionCallsValidator out into someplace potentially more reusable.

** I realize by example above would also require `expr.__str__` -- baby steps.

Reply:
This makes sense will do. Though I will call the intermediate ExpressionValidationException instead of QueryValidationException since it deals with one individual expression.

About expr.__str__, yeah we have it on. the backlog. Right now the default __Str__ implementation is not the most readable representation, but it still include all the important details so it is better than nothing.

Reply:
done